src/Stratis.StratisDnsD/Program.cs

```diff
@@ -66,25 +60,30 @@ public static void Main(string[] args)
                         .AddRPC()
                         .UseDns()
                         .Build();
-
-                    // Run node.
-                    if (node != null)
-                        await node.RunAsync();
                 }
                 else
                 {
                     // Build the Dns node.
-                    IFullNode node = new FullNodeBuilder()
+                    node = new FullNodeBuilder()
                         .UseNodeSettings(nodeSettings)
                         .UsePosConsensus()
                         .UseApi()
                         .AddRPC()
                         .UseDns()
                         .Build();
+                }
+
+                // Run node.
+                if (node != null)
+                {
+                    DnsSettings dnsSettings = node.NodeService<DnsSettings>();
```

Comment:
Why not do `DnsSettings dnsSettings = new DnsSettings().Load(nodeSettings);` like it was before?
Also why are you delaying applying the DNS settings? Is it not better to apply those first and then try to run the node?

Reply:
It is not appropriate to load the DNS settings (and invoke corresponding exceptions) in the case where we may simply be displaying command-line help. 

The main reason for the apparent departure from the norm is the extra code required for throwing the `ArgumentException`. It requires peeking ahead instead of waiting for `RunAsync` to load the DNS settings.